lambdas/process-journey-event/src/main/resources/statemachine/journey-maps/new-p2-identity.yaml

```diff
@@ -515,9 +521,17 @@ states:
         targetState: EVALUATE_GPG45_SCORES
       enhanced-verification:
         targetState: MITIGATION_02_OPTIONS_WITH_F2F
+        auditEvents:
+          - IPV_MITIGATION_START
+        auditContext:
+          mitigationType: enhanced-verification
         checkIfDisabled:
           f2f:
             targetState: MITIGATION_02_OPTIONS
+            auditEvents:
+              - IPV_MITIGATION_START
+            auditContext:
+              mitigationType: enhanced-verification
```

Comment:
Interested in opinions on this syntax, I've tried to make this as flexible as possible while staying concise, but we could streamline in a few ways:
- `auditEvents` could be singular - but that means you can't record multiple events at the same time, which we might need to do for COI
- `auditContext` could be a simple string - but an object allows much more specificity
- we could 'inherit' audit events for these disabled/feature checks to reduce duplication - at the moment that's desirable, but it wouldn't have been while these mitigations were behind a feature flag (for example)

Reply:
Plural `auditEvents` and an object for `auditContext` sounds sensible but yeah inheriting seems a bit risky especially if it's not obvious how it works

Reply:
Part of me is wondering if we want the same audit context tied to all the audit events? Or if there is any benefit in each event having its own context? Something like:
```
enhanced-verification:
  targetState: MITIGATION_02_OPTIONS_WITH_F2F
  auditEvents:
    - event: IPV_MITIGATION_START
      auditContext:
        mitigationType: enhanced-verification
    - event: SOMETHING ELSE
      auditContext:
        other: stuff
```

I supposed it doesn't really matter which event consumes which part of the context - it just feels a bit "loose" like that.

Reply:
Yeah, I think that's a good question - that would be more precise, but a little more verbose. Part of my motivation was to try and make the 'simple' cases as easy as possible.

I wonder if there's a compromise at:
```
enhanced-verification:
  targetState: MITIGATION_02_OPTIONS_WITH_F2F
  auditEvents:
    - event: IPV_MITIGATION_START
      mitigationType: enhanced-verification
    - event: SOMETHING_ELSE
```

It'd need a slightly more interesting type (to allow arbitrary additional parameters), but should be doable.

Reply:
In practice I think we're unlikely to often need multiple events in the same place, and even less often to need context on both.

Reply:
Maybe we do that refactor if we need it. I think the bag of stuff approach is proabbly fine, especially if we're actually going to be limited in where we have multiple events.

Reply:
I missed the bit about inheriting - but I also agree. Something would get missed and we'd send the wrong thing.